cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -0,0 +1,391 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | An API for driving on-chain poll for SPOs.
+--
+-- Polls are done on-chain through transaction metadata and authenticated via
+-- stake pool credentials (either VRF public key or Ed25519 cold key).
+--
+-- The goal is to gather opinions on governance matters such as protocol
+-- parameters updates. This standard is meant to be an inclusive interim
+-- solution while the work on a larger governance framework such as
+-- CIP-1694 continues.
+module Cardano.Api.Governance.Poll(
+    -- * Type Proxies
+    AsType(..),
+
+    -- * Types
+    GovernancePoll (..),
+    GovernancePollAnswer (..),
+    GovernancePollWitness (..),
+
+    -- * Errors
+    GovernancePollError (..),
+    renderGovernancePollError,
+
+    -- * Functions
+    hashGovernancePoll,
+    signPollAnswerWith,
+    verifyPollAnswer,
+  ) where
+
+import           Control.Arrow (left)
+import           Control.Monad (unless, when)
+import qualified Data.ByteString as BS
+import           Data.Either.Combinators (maybeToRight)
+import qualified Data.Map.Strict as Map
+import           Data.String (IsString(..))
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import           Data.Word (Word64)
+
+import           Cardano.Api.HasTypeProxy
+import           Cardano.Api.Hash
+import           Cardano.Api.SerialiseCBOR
+import           Cardano.Api.SerialiseRaw
+import           Cardano.Api.SerialiseTextEnvelope
+import           Cardano.Api.SerialiseUsing
+import           Cardano.Api.TxMetadata
+import           Cardano.Api.Utils
+
+import           Cardano.Binary (DecoderError(..))
+import           Cardano.Ledger.Crypto (HASH, StandardCrypto, VRF)
+import           Cardano.Ledger.Keys (KeyRole(..), SignedDSIGN, SignKeyDSIGN,
+                   SignKeyVRF, VKey(..), VerKeyVRF, signedDSIGN, verifySignedDSIGN)
+
+import qualified Cardano.Crypto.DSIGN as DSIGN
+import           Cardano.Crypto.Hash (hashFromBytes, hashToBytes, hashWith)
+import qualified Cardano.Crypto.Hash as Hash
+import           Cardano.Crypto.Util (SignableRepresentation(..))
+import qualified Cardano.Crypto.VRF as VRF
+
+-- | Associated metadata label as defined in CIP-XXXX
```

Comment:
What is `CIP-XXXX`?

Reply:
Ah, so `CIP-0094`?

Reply:
Most likely, though numbers hasn't been assigned yet. I think I can get CIP editors to assign 94 to that one 😏